### PR TITLE
Refactor error handling

### LINKED ISSUE #400: [Portal] Invent a better abstraction for showing errors
We can borrow some ideas from the prior art by Terry. The main goals are

1. Avoid conversions. They do not add any value.
2. Be declarative. Imperative errors classification is long and error-prone.
3. Support classified errors and unclassified errors. Classified errors should typically be shown near the control. Unclassified errors should be presented globally, like at the top of the screen and as a dialog.

### PR DESCRIPTION
**Pull request by axolotl0212**
fix #400 

- Skip transforming error to Violation
- Split error handling to two layers

TODO: implement this to all screen

### CODE REVIEW

**Comment by axolotl0212** [comment]
@louischan-oursky updated

File: portal/src/error/duplicated.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface APIDuplicatedIdentityError {
+  errorName: string;
+  reason: "DuplicatedIdentity";
```
Comment (louischan-oursky) [inline portal/src/error/duplicated.ts:3]: This error is going to be removed in https://github.com/authgear/authgear-server/pull/488 https://github.com/authgear/authgear-server/pull/488/commits/a14e7eb09ca0578e438d3a418afcc2524423f24c

**Review by louischan-oursky — COMMENTED** [review]
The API looks much cleaner now!

File: portal/src/error/duplicated.ts (hunk 1 above)
Comment (axolotl0212) [reply portal/src/error/duplicated.ts:3]: It's already removed, will delete this file

**Comment by axolotl0212** [comment]
@louischan-oursky updated and rebased

**Pull request merged**